shap/plots/_bar.py

```diff
@@ -273,33 +278,33 @@ def bar(shap_values, max_display=10, order=Explanation.abs, clustering=None, clu
             except Exception:
                 pass # features[i] must not be a number
 
-    pl.gca().xaxis.set_ticks_position('bottom')
-    pl.gca().yaxis.set_ticks_position('none')
-    pl.gca().spines['right'].set_visible(False)
-    pl.gca().spines['top'].set_visible(False)
+    ax.xaxis.set_ticks_position('bottom')
+    ax.yaxis.set_ticks_position('none')
+    ax.spines['right'].set_visible(False)
+    ax.spines['top'].set_visible(False)
     if negative_values_present:
-        pl.gca().spines['left'].set_visible(False)
-    pl.gca().tick_params('x', labelsize=11)
+        ax.spines['left'].set_visible(False)
+    ax.tick_params('x', labelsize=11)
 
-    xmin,xmax = pl.gca().get_xlim()
-    ymin,ymax = pl.gca().get_ylim()
+    xmin,xmax = ax.get_xlim()
+    ymin,ymax = ax.get_ylim()
 
     if negative_values_present:
-        pl.gca().set_xlim(xmin - (xmax-xmin)*0.05, xmax + (xmax-xmin)*0.05)
+        ax.set_xlim(xmin - (xmax-xmin)*0.05, xmax + (xmax-xmin)*0.05)
```

Comment:
nit: could be marginally clearer to refactor `(xmax-xmin)*0.05` out as a variable called padding or buffer maybe?

Reply:
Good shout. Incidentally there is probably a lot more we could refactor in this function; but I'll aim to keep the scope of this PR fairly small - demonstrating the main change we want to make to the plotting API.

Reply:
yes, the function is super long, would be good to break it up in the future maybe.

Also, more controversial, but why import matplotlib as pl instead of plt (which I think is pretty standard)?